syft/src/main/java/org/openmined/syft/data/fetchers/BaseDatasetFetcher.kt

```diff
@@ -0,0 +1,30 @@
+package org.openmined.syft.data.fetchers
+
+import org.openmined.syft.data.Dataset
+import org.pytorch.IValue
+import org.pytorch.Tensor
+
+open class BaseDatasetFetcher(val dataset: Dataset) : Fetcher {
```

Comment:
From the looks of it this class is only responsible for collating the independent data points into batch. So why do we need a separate class for this? Can't there be a collating function in the dataloader class? I agree this collating method should be overrideable and the user can simply pass this method as argument if he wants